tests/integration/awslambda/test_lambda_api.py

```diff
@@ -523,6 +528,221 @@ def test_list_functions(self, lambda_client, create_lambda_function, lambda_su_r
         snapshot.match("list_default", list_default)
 
 
+@pytest.mark.skipif(is_old_provider(), reason="focusing on new provider")
+class TestLambdaImages:
+    @pytest.fixture(scope="class")
+    def login_docker_client(self, ecr_client):
+        if not is_aws_cloud():
+            return
+        auth_data = ecr_client.get_authorization_token()
+        # if check is necessary since registry login data is not available at LS before min. 1 repository is created
+        if auth_data["authorizationData"]:
+            auth_data = auth_data["authorizationData"][0]
+            decoded_auth_token = str(
+                base64.decodebytes(bytes(auth_data["authorizationToken"], "utf-8")), "utf-8"
+            )
+            username, password = decoded_auth_token.split(":")
+            DOCKER_CLIENT.login(
+                username=username, password=password, registry=auth_data["proxyEndpoint"]
+            )
+
+    @pytest.fixture(scope="class")
+    def test_image(self, ecr_client, login_docker_client):
+        repository_names = []
+        image_names = []
+
+        def _create_test_image(base_image: str):
+            if is_aws_cloud():
+                repository_name = f"test-repo-{short_uid()}"
+                repository_uri = ecr_client.create_repository(repositoryName=repository_name)[
+                    "repository"
+                ]["repositoryUri"]
+                image_name = f"{repository_uri}:latest"
+                repository_names.append(repository_name)
+            else:
+                image_name = f"test-image-{short_uid()}:latest"
+            image_names.append(image_name)
+
+            DOCKER_CLIENT.pull_image(base_image)
+            DOCKER_CLIENT.tag_image(base_image, image_name)
+            if is_aws_cloud():
+                DOCKER_CLIENT.push_image(image_name)
+            return image_name
+
+        yield _create_test_image
+
+        for image_name in image_names:
+            try:
+                DOCKER_CLIENT.remove_image(image=image_name, force=True)
+            except Exception as e:
+                LOG.debug("Error cleaning up image %s: %s", image_name, e)
+
+        for repository_name in repository_names:
+            try:
+                image_ids = ecr_client.list_images(repositoryName=repository_name)["imageIds"]
+                ecr_client.batch_delete_image(repositoryName=repository_name, imageIds=image_ids)
+                ecr_client.delete_repository(repositoryName=repository_name)
+            except Exception as e:
+                LOG.debug("Error cleaning up repository %s: %s", repository_name, e)
```

Comment:
```suggestion
            try:
                image_ids = ecr_client.list_images(repositoryName=repository_name).get("imageIds", [])
                if image_ids:
                    call_safe(ecr_client.batch_delete_image, kwargs={"repositoryName": repository_name, "imageIds": image_ids})
                ecr_client.delete_repository(repositoryName=repository_name)
            except Exception as e:
                LOG.debug("Error cleaning up repository %s: %s", repository_name, e)
```

to avoid potentially  leaking the repo

Reply:
If the batch delete image operation fails for some reason, the delete repository will not go through (due to images present in the repo) anyway, but I added it!